pxe/common_files/disk_imaging.sh

```
@@ -314,11 +339,253 @@ function modify_grub_cfg() {
+	# Trade-off: this is ESP-blind. A legitimately separate install of the
+	# same distro on another disk in the same machine shares the label and
+	# would be deleted too. That is acceptable for whole-node provisioning,
+	# where the imager owns the box, but it is the reason this is scoped to
+	# an exact label rather than a substring.
+	#
+	# "-B -L" needs efibootmgr >= 18 (delete_label() does not exist in 17)
+	# and exits non-zero when nothing matched, which is the normal case on a
+	# first install, so the failure is logged rather than treated as fatal.
+	echo "Removing any existing EFI boot entries labelled $label" | tee $log_output
+	delete_out=$(efibootmgr -B -L "$label" 2>&1)
+	if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)